tests/tools.vitruv.applications.pcmjava.tests/src/tools/vitruv/applications/pcmjava/tests/pcm2java/repository/CompositeComponentMappingTransformationTest.xtend

```diff
@@ -0,0 +1,171 @@
+package tools.vitruv.applications.pcmjava.tests.pcm2java.repository
+
+import java.util.List
+import org.junit.jupiter.api.Test
+import org.palladiosimulator.pcm.repository.BasicComponent
+import org.palladiosimulator.pcm.repository.CompositeComponent
+import tools.vitruv.applications.pcmjava.tests.pcm2java.Pcm2JavaTestUtils
+import tools.vitruv.applications.pcmjava.tests.pcm2java.Pcm2JavaTransformationTest
+import tools.vitruv.applications.pcmjava.tests.pcm2java.javahelper.ConstructorArguments
+import tools.vitruv.applications.pcmjava.tests.pcm2java.javahelper.JavaClassBuilder
+import tools.vitruv.applications.pcmjava.tests.pcm2java.javahelper.JavaInterfaceBuilder
+
+import static tools.vitruv.applications.pcmjava.tests.pcm2java.PcmCreatorsUtil.*
+import static tools.vitruv.applications.pcmjava.tests.pcm2java.javahelper.JavaCreatorsUtil.*
+
+import static extension edu.kit.ipd.sdq.commons.util.java.lang.IterableUtil.*
+import static extension tools.vitruv.applications.pcmjava.tests.pcm2java.PcmQueryUtil.*
+
+class CompositeComponentMappingTransformationTest extends Pcm2JavaTransformationTest {
+	
+	@Test
+	def void testCreateCompositeComponent() {
+		createRepostory(Pcm2JavaTestUtils.REPOSITORY_NAME)
+		
+		changePcmView[
+			modifySingleRepository[
+				components__Repository += createCompositeComponent(Pcm2JavaTestUtils.COMPOSITE_COMPONENT_NAME)	
+			]
+		]
+		
+		validateJavaView[
+			val expectedCompilationUnit = new JavaClassBuilder( 
+				Pcm2JavaTestUtils.COMPOSITE_COMPONENT_NAME + Pcm2JavaTestUtils.IMPL_SUFIX, 
+				Pcm2JavaTestUtils.REPOSITORY_NAME + "." + Pcm2JavaTestUtils.COMPOSITE_COMPONENT_NAME
+			).build
+			
+			assertCompilationUnits(List.of(expectedCompilationUnit))
+		]
+	}
+	
+	@Test
+	def void testAddProvidedRoleToCompositeComponent() {
+		this.addRepositoryAndCompositeComponent()
+		changePcmView[
+			modifySingleRepository[
+				interfaces__Repository += createOperationInterface(Pcm2JavaTestUtils.INTERFACE_NAME)
+			]
+		]
+		
+		changePcmView[
+			modifySingleRepository[
+				var operationInterface = it.claimOperationInterface(Pcm2JavaTestUtils.INTERFACE_NAME)
+				var compositeComponent = claimComponent(Pcm2JavaTestUtils.COMPOSITE_COMPONENT_NAME)
+				compositeComponent.providedRoles_InterfaceProvidingEntity += createOperationProvidedRole(operationInterface, compositeComponent)
+			]
+		]
```

Comment:
Why do you modify the views in two steps? This pattern occurs frequently. If you want to preserve this two step behavior - as it may mirror more closely how an actual developer might work or for whatever reason - you should add some validation step between the two modifications. For our basic tests however, for me it would totally suffice just having one `changePcmView` where everything is modified.

Reply:
I followed the arrange-act-assert pattern in all of the unit tests. The second `changePcmView` makes clear which of the behavior is actually tested.